kafka/integration_test.go

```diff
@@ -2684,6 +2684,93 @@ func (its *IntegrationTestSuite) TestAdminClient_UserScramCredentials() {
 	}
 }
 
+// Tests ListOffsets API which describes
+// the offset of a TopicPartition corresponding to the OffsetSpec provided.
+func (its *IntegrationTestSuite) TestAdminClient_ListOffsets() {
+	t := its.T()
+	bootstrapServers := testconf.Brokers
+	rand.Seed(time.Now().Unix())
+	assert := its.Assert()
+
+	// Create a new AdminClient.
+	a := createAdminClient(t)
+	defer a.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	topicPartitionOffsets := make(map[TopicPartition]OffsetSpec)
+	Topic := fmt.Sprintf("%s-%d", testconf.Topic, rand.Int())
+
+	topics := []TopicSpecification{TopicSpecification{Topic: Topic, NumPartitions: 1, ReplicationFactor: 1}}
+	createTopicResult, createTopicError := a.CreateTopics(ctx, topics)
+	assert.Nil(createTopicError, "Create Topics should not fail.")
+	assert.Equal(createTopicResult[0].Error.Code(), ErrNoError, "Create Topics Error Code should be 0.")
+
+	p, err := NewProducer(&ConfigMap{"bootstrap.servers": bootstrapServers})
+	assert.Nil(err, "Unable to create Producer.")
+	defer p.Close()
+
+	timestamp := time.Now()
+	t1 := timestamp.Add(time.Second * 100)
+	t2 := timestamp.Add(time.Second * 300)
+	t3 := timestamp.Add(time.Second * 200)
+
+	p.Produce(&Message{
+		TopicPartition: TopicPartition{Topic: &Topic, Partition: 0},
+		Value:          []byte("Message-1"),
+		Timestamp:      t1,
+	}, nil)
+
+	p.Produce(&Message{
+		TopicPartition: TopicPartition{Topic: &Topic, Partition: 0},
+		Value:          []byte("Message-2"),
+		Timestamp:      t2,
+	}, nil)
+
+	p.Produce(&Message{
+		TopicPartition: TopicPartition{Topic: &Topic, Partition: 0},
+		Value:          []byte("Message-3"),
+		Timestamp:      t3,
+	}, nil)
+
+	p.Flush(5 * 1000)
+
+	tp1 := TopicPartition{Topic: &Topic, Partition: 0}
+	topicPartitionOffsets[tp1] = EarliestOffsetSpec
+	var results ListOffsetsResult
+	results, err = a.ListOffsets(ctx, topicPartitionOffsets, SetAdminIsolationLevel(ReadCommitted))
+	assert.Nil(err, "ListOffsets should not fail.")
+
+	for _, info := range results.Results {
+		assert.Equal(info.Error.Code(), ErrNoError, "Error code should be 0.")
+		assert.Equal(info.Offset, int64(0), "Offset should be 0.")
+	}
+
+	topicPartitionOffsets[tp1] = LatestOffsetSpec
+	results, err = a.ListOffsets(ctx, topicPartitionOffsets, SetAdminIsolationLevel(ReadCommitted))
+	assert.Nil(err, "ListOffsets should not fail.")
+
+	for _, info := range results.Results {
+		assert.Equal(info.Error.Code(), ErrNoError, "Error code should be 0.")
+		assert.Equal(info.Offset, int64(3), "Offset should be 3.")
+	}
+
+	topicPartitionOffsets[tp1] = OffsetSpec(MaxTimestampOffsetSpec)
+	results, err = a.ListOffsets(ctx, topicPartitionOffsets, SetAdminIsolationLevel(ReadCommitted))
+	assert.Nil(err, "ListOffsets should not fail.")
+
+	for _, info := range results.Results {
+		assert.Equal(info.Error.Code(), ErrNoError, "Error code should be 0.")
```

Comment:
The message should be changed too, not just the assert. 
Like
```suggestion
		assert.Equal(info.Error.Code(), ErrNoError, "Error code should be ErrNoError.")
```

Reply:
At all three places.